### PR TITLE
feat: add structured output support via --json-schema argument

### PR DESCRIPTION
**Pull request by bogini**
Add support for Agent SDK structured outputs.

**Usage**: Pass `--json-schema` flag in `claude_args` input  
**Output**: `structured_output` - JSON string with all fields  
**Access**: `fromJSON(steps.id.outputs.structured_output).field_name`

## Example

```yaml
- id: analyze
  uses: anthropics/claude-code-action@v1
  with:
    prompt: "Analyze CI logs for flakiness"
    claude_args: |
      --json-schema '{"type":"object","properties":{"is_flaky":{"type":"boolean"}}}'

- if: fromJSON(steps.analyze.outputs.structured_output).is_flaky == true
  run: gh workflow run CI --ref ${{ github.ref }}
```

**Docs**: https://docs.claude.com/en/docs/agent-sdk/structured-outputs

**Example**: `examples/test-failure-analysis.yml`

### CODE REVIEW

File: base-action/src/index.ts [hunk 1]
```diff
@@ -28,8 +28,17 @@ async function run() {
       promptFile: process.env.INPUT_PROMPT_FILE || "",
     });
 
+    // Build claudeArgs with JSON schema if provided
+    let claudeArgs = process.env.INPUT_CLAUDE_ARGS || "";
+
+    // Add JSON schema if specified (no escaping - parseShellArgs handles it)
+    if (process.env.JSON_SCHEMA) {
+      // Wrap in single quotes for parseShellArgs
+      claudeArgs += ` --json-schema '${process.env.JSON_SCHEMA}'`;
```
[the remark this answers is not in the document]
Comment (bogini) [reply base-action/src/index.ts]: This applies to all inputs, not just the one added in this PR and is not a concern since the inputs are provided by the GH workflow owner.

File: base-action/src/run-claude.ts [hunk 2]
```diff
@@ -122,6 +122,48 @@ export function prepareRunConfig(
...
+/**
+ * Parses structured_output from execution file and sets GitHub Action outputs
+ * Only runs if json_schema was explicitly provided by the user
+ * Exported for testing
+ */
+export async function parseAndSetStructuredOutputs(
+  executionFile: string,
+): Promise<void> {
+  try {
+    const content = await readFile(executionFile, "utf-8");
```
[the remark this answers is not in the document]
Comment (bogini) [reply base-action/src/run-claude.ts:134]: Not needed.

File: base-action/test/structured-output.test.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  test("should throw error with malformed JSON", async () => {
+    await writeFile(TEST_EXECUTION_FILE, "{ invalid json");
+
+    await expect(
+      parseAndSetStructuredOutputs(TEST_EXECUTION_FILE),
+    ).rejects.toThrow();
+  });
+
+  test("should throw error when file does not exist", async () => {
+    await expect(
```
[the remark this answers is not in the document]
Comment (bogini) [reply base-action/test/structured-output.test.ts:143]: Minor edge case

**Review by ashwin-ant — DISMISSED** [review]

**Review by ashwin-ant — APPROVED** [review]
